docs/modules/ROOT/pages/contributors/naming-conventions.adoc

```diff
@@ -0,0 +1,212 @@
+= Naming Conventions
+
+This document lists the naming conventions applied in different parts
+of the KubeArchive codebase.
+
+== Helm Chart
+
+The Helm Chart templates are located in different directories that match the different
+components of KubeArchive:
+
+* Sink
+* API Server
+* Operator
+* Database
+
+The common resources are outside these folders.
+
+The templated variables are located in [filename]`kubearchive/chars/kubearchive/values.yaml`
+following the same categories.
+
+=== Conventions
+
+* Use the
+link:https://helm.sh/docs/chart_template_guide/builtin_objects/[Helm builtin variables]
+when possible.
++
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  # BAD - name: kubearchive-proxy
+  name: {{ .Release.Name }}-proxy
+subjects:
+  - kind: ServiceAccount
+    # BAD - name: kubearchive-operator
+    name: {{ .Release.Name }}-operator
+    # BAD - namespace: kubearchive
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  # BAD - name: kubearchive-proxy
+  name: {{ .Release.Name }}-proxy
+----
+
+* Do not add the `Kind` of the resource as part of the `name`.
+We encourage short non-redundant readable names.
++
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  # BAD - name: {{ .Release.Name }}-test-service-account
+  name: {{ .Release.Name }}-test
+----
+
+* Do not create variables with the same value, rename the variable if needed
+and reuse it.
+We want to keep the list of variables as short as possible.
++
+.Bad
+.values.yaml
+[source,yaml]
+----
+apiServer:
+  # BAD:
+  # deployment: {{ .Release.Name }}-api-server
+  # serviceAccount: {{ .Release.Name}}-api-server
+  # service: {{ .Release.Name }}-api-server
+  # role: {{ .Release.Name }}-api-server
+  # roleBinding: {{ .Release.Name }}-api-server
+  # GOOD:
+  name: {{ .Release.Name }}-api-server
+----
+
+* Do not set the `namespace` metadata field if it's going to have the same
+value as `{{ .Release.Namespace }}`. It's redundant and may introduce an error.
++
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-test
+  # BAD - namespace: {{ .Release.Namespace }}
+----
+
+* Extract the variables that the user may want to change to the [filename]`values.yaml`
+file.
+If you want to reuse a variable that is not meant to be exposed, go to the next point.
```

Comment:
I don't think this sentence is necessary since, like it says, this issue is addressed in the next bullet

Reply:
Removed